### PR TITLE
🐛 no longer push outputs when service starts

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Only trigger change events when files are: create, moved, deleted.
Skip propagating events when files permission or ownership is changed.

We are not interested in triggering uploads in these situations. Such was the case for the services which start and change ownership of their outputs and inputs directories. This will guarantee that services stop pushing their outputs when started.

## Related issue/s

- fixes https://github.com/ITISFoundation/osparc-issues/issues/730
- https://github.com/ITISFoundation/osparc-issues/issues/638

## Checklist

- [x] Unit tests for the changes exist

### CODE REVIEW

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_core.py [hunk 1]
```diff
@@ -126,8 +127,9 @@ def set_enabled(self, is_enabled: bool) -> None:
         self._is_enabled = is_enabled
 
     def _invoke_push_directory(self) -> None:
-        # wrapping the function call in the object
-        # helps with testing, it is simplet to mock
+        if not self._is_enabled:
+            return
```
Comment (sanderegg) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_core.py:131]: very minor: not sure I get the point of moving this if from the ```on_any_event``` it was nice over there no?

File: services/dynamic-sidecar/tests/unit/test_modules_directory_watcher.py [hunk 2]
```diff
@@ -72,7 +69,7 @@ async def test_run_observer(
     # generates the first event chain
     await _generate_event_burst(tmp_path)
 
-    await asyncio.sleep(2)
+    await _wait_for_events_to_trigger()
```
Comment (sanderegg) [inline services/dynamic-sidecar/tests/unit/test_modules_directory_watcher.py:72]: 👍

File: services/dynamic-sidecar/tests/unit/test_modules_directory_watcher.py [hunk 3]
```diff
@@ -85,3 +82,29 @@ async def test_run_observer(
...
+
+    await _wait_for_events_to_trigger()
+    assert patch_directory_watcher.call_count == 1
+
+    # apply an attribute change
+    file_path_1.chmod(0o744)
+
+    await _wait_for_events_to_trigger()
+    # same call count as before, event was ignored
+    assert patch_directory_watcher.call_count == 1
```
Comment (sanderegg) [inline services/dynamic-sidecar/tests/unit/test_modules_directory_watcher.py:110]: very nice!

**Review by sanderegg — APPROVED** [review]
this should have quite an effect on start times.

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_watchdog_extentions.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        InotifyConstants.IN_DELETE_SELF,
+        InotifyConstants.IN_DONT_FOLLOW,
+        InotifyConstants.IN_CLOSE_WRITE,
+    ],
+)
+
+
+class _ExtendedInotifyBuffer(InotifyBuffer):
+    def __init__(self, path, recursive=False):  # pylint:disable=super-init-not-called
+        BaseThread.__init__(self)  # pylint:disable=non-parent-init-called
```
Comment (pcrespov) [inline services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_watchdog_extentions.py:27]: As pylint warns, this is indeed a strange setup. I [see](https://github.com/gorakhargosh/watchdog/blob/78adc2f8b61858be28e8056b7a88f4a4ae7acab7/src/watchdog/observers/inotify_buffer.py) that you actually want to only override ``InotifyBuffer.__init__`` in order to change ``self._inotify`` before it ``self.start()``.
i wonder if there is a better way using ``__new__`` ... i need to think about it and will come back to you

**Review by pcrespov — APPROVED** [review]

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_watchdog_extentions.py (hunk 4 above)
Comment (GitHK) [reply services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_watchdog_extentions.py:27]: Yes, the internals of this library do not give me much flexibility. This was the cleanest way to do it. I'd appreciate further input here, on how to make make it more clean.

File: services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_core.py (hunk 1 above)
Comment (GitHK) [reply services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/directory_watcher/_core.py:131]: I did a lot of tries and at some point this function was called from different handlers. I left it here, where I think it should be located.

**Review by elisabettai — APPROVED** [review]
Looking forward to this, should help opening (big) studies faster!

**Pull request merged**